File: webapp/src/model/ticker-event.ts
import { TickerEventType } from "common/common/enums";

export class TickerEvent {
  time: Date;
  pair: string;
  price: number;
  type: TickerEventType;
  size: number;
  exchange: string;

  constructor(data: {time: Date, pair: string, price: number, type: TickerEventType, size: number, exchange: string} ) {
    this.time = data.time;
    this.pair = data.pair;
    this.price = data.price;
    this.type = data.type;
    this.size = data.size;
    this.exchange = data.exchange;
  }
}

File: message-listener-java/domain/src/main/java/com/ticker/message/listener/domain/entities/TickerEvent.java
package com.ticker.message.listener.domain.entities;

import com.ticker.message.listener.common.enums.TickerEventType;
import lombok.Data;
import lombok.experimental.Accessors;

import java.util.Date;

@Data
@Accessors(chain = true)
public class TickerEvent {
    private Date time;
    private String pair;
    private Double price;
    private TickerEventType type;
    private String exchange;
}
